node/pkg/p2p/p2p.go

```diff
@@ -355,24 +366,84 @@ func Run(params *RunParams) func(ctx context.Context) error {
 			panic(err)
 		}
 
-		th, err := ps.Join(topic)
-		if err != nil {
-			return fmt.Errorf("failed to join topic: %w", err)
+		// These will only be non-nil if the application plans to listen for or publish to that topic.
+		var controlPubsubTopic, attestationPubsubTopic, vaaPubsubTopic *pubsub.Topic
+		var controlSubscription, attestationSubscription, vaaSubscription *pubsub.Subscription
+
+		// Set up the control channel. ////////////////////////////////////////////////////////////////////
+		if params.nodeName != "" || params.gossipControlSendC != nil || params.obsvReqSendC != nil || params.obsvReqC != nil || params.signedGovCfg != nil || params.signedGovSt != nil {
+			controlTopic := fmt.Sprintf("%s/%s", params.networkID, "control")
+			logger.Info("joining the control topic", zap.String("topic", controlTopic))
+			controlPubsubTopic, err = ps.Join(controlTopic)
+			if err != nil {
+				return fmt.Errorf("failed to join the control topic: %w", err)
+			}
+
+			defer func() {
+				if err := controlPubsubTopic.Close(); err != nil && !errors.Is(err, context.Canceled) {
+					logger.Error("Error closing the control topic", zap.Error(err))
+				}
+			}()
+
+			if params.obsvReqC != nil || params.signedGovCfg != nil || params.signedGovSt != nil {
+				logger.Info("subscribing to the control topic", zap.String("topic", controlTopic))
+				controlSubscription, err = controlPubsubTopic.Subscribe(pubsub.WithBufferSize(P2P_SUBSCRIPTION_BUFFER_SIZE))
+				if err != nil {
+					return fmt.Errorf("failed to subscribe to the control topic: %w", err)
+				}
+				defer controlSubscription.Cancel()
```

Comment:
This is different than the other 2 channels?  This one doesn't have a defer func to close?

Reply:
That's because `pubsub.Subscription.Cancel` doesn't return a value, so the defer just calls it directly. That's the way the subscribe for the old topic was too.